### PR TITLE
Cotco 249

### PR DESCRIPTION
**Pull request by andreicostin**

### CODE REVIEW

File: app/assets/javascript/custom.js [hunk 1]
```diff
@@ -17,80 +17,38 @@ var copyCode = function(codeContainer) {
...
 
-    if (diff > 0) {
-      setInterval(timer, that.granularity);
-      if (diff < 60) {
+  function myClock() {
+    document.querySelector(".countdown").innerHTML = --c;
+    if (c == 0) {
+      clearInterval(myTimer);
+      alert("Expired");
+    } else if (c < 61) {
```
Comment (peppe-hmrc) [inline app/assets/javascript/custom.js]: I'd imagine this means that 1 minute before expiration a message will be shown.
My first question is if `<= 60` would be more clear but also would make sense to have these values in a sort of configuration file?
I'm not a javascript developer so ignore me if I'm saying something wrong :)

File: app/assets/javascript/custom.js [hunk 2]
```diff
@@ -103,39 +61,11 @@ var errors = {
...
-      countdown.textContent = minutesInSeconds + Number(seconds);
-    }
-
-    $(".extend-button").on("click", function() {
-        var extension = 600 -  Number($('.countdown').text());
-        countdownRemoveMessage();
-        extend(extension);
-    });
-  }
+  countdown(600);
```
Comment (peppe-hmrc) [inline app/assets/javascript/custom.js]: same here for the 600 .... wondering how difficult would be to move from 10minutes to 5minutes

File: app/assets/javascript/custom.js (hunk 1 above)
Comment (udinobi) [reply app/assets/javascript/custom.js]: Me and Andrei already had a discussion passing config params from the backend. The problem with that solution is that to pass them to javascript you need a script tag in the template and it looks like the GDPR are not happy with that.

Comment (peppe-hmrc) [reply app/assets/javascript/custom.js]: but something like global variables? 
Just a place where these values are easy to manipulate and change if needed, but again I'm talking without knowing the language

**Comment by andreicostin** [comment]
Yes, it makes sense to use store these in a global object. Will do that tomorrow - tidying up my coding.
  

  

  
  

  
  
>

**Comment by andreicostin** [comment]
Changes made! please review!

**Pull request merged**